File: uva-online-judge/cpp/299 - Train Swapping.cpp
#include <iostream>

using namespace std;

int main()
{
    int N;
    cin >> N;
    for (int i = 0; i < N; i++)
    {
        int L, count = 0;
        cin >> L;
        int numbers[L];
        for (int j = 0; j < L; j++)
        {
            cin >> numbers[j];
        }
        while (true)
        {
            for (int j = 0; j < L - 1; j++)
            {
                if (numbers[j] > numbers[j + 1])
                {
                    count++;
                }
            }
            int counter = 0;
            for (int j = 0; j < L - 1; j++)
            {
                if (numbers[j] < numbers[j + 1])
                {
                    counter++;
                }
            }
            if (counter == L - 1)
            {
                break;
            }
        }
        cout << "Optimal train swapping takes " << count << " swaps.\n";
    }

    return 0;
}

File: uva/299 - Train Swapping.py
t = int(input())
for i in range(t):
    value = int(input())
    numbers = list(map(int, input().strip().split()))
    count = 0
    length = len(numbers)
    while True:
        for item in range(length - 1):
            if numbers[item] > numbers[item + 1]:
                count = count + 1
                numbers[item], numbers[item + 1] = numbers[item + 1], numbers[item]
        counter = 0
        for x in range(length - 1):
            if numbers[x] < numbers[x + 1]:
                counter = counter + 1
        if counter == len(numbers) - 1:
            break
    print(f"Optimal train swapping takes {count} swaps.")
